Remove deprecated -f flag on docker tag

yongtang opened the pull request:
The -f flag on docker tag has been deprecated in docker 1.10 and is expected to be removed in docker 1.12.
This fix removed the -f flag on docker tag and also updated deprecated.md.

**\- Description for the changelog**

The deprecated `-f` flag on `docker tag` has been removed.

NOTE: A separate pull request for engine-api has been opened to cover the related changes:
https://github.com/docker/engine-api/pull/239

Signed-off-by: Yong Tang yong.tang.github@outlook.com

yongtang committed:
Remove deprecated -f flag on docker tag

The -f flag on docker tag has been deprecated in docker 1.10 and
is expected to be removed in docker 1.12.

This fix removed the -f flag on docker tag and also updated
deprecated.md.

NOTE: A separate pull request for engine-api has been opened to
cover the related changes.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

thaJeztah commented:
thanks! can you check the bash/zsh completion scripts as well? not sure if they still contain this flag

thaJeztah commented:
also /cc @dnephin in case your PR needs to be updated

yongtang commented:
Thanks @thaJeztah. I checked `contrib/completion/bash/docker` and `contrib/completion/zsh/_docker`, didn't find the reference of `-f/-force` for `docker tag`.

albers commented:
@thaJeztah `-f` was removed from the completions in #19642, so we are clean here.

thaJeztah commented:
That's for checking! (Was typing on my phone, so wasn't able to check myself)

LGTM

yongtang committed:
Vendor engine-api to 6facb3f3c38717b8f618dcedc4c8ce20d1bfc61e

This fix updates engine-api to 6facb3f3c38717b8f618dcedc4c8ce20d1bfc61e.

This fix is related to #23090.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

thaJeztah commented:
@yongtang https://github.com/docker/engine-api/issues/239 was merged, so you can update the vendored engine-API files

yongtang commented:
Thanks @thaJeztah the pull request has been updated to include the engine-api changes. Let me know if there are any issues.

thaJeztah commented:
thanks! LGTM

vdemeester commented:
LGTM 🐼

vdemeester commented:
Moving this to docs-review. Looks like at least https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.24.md#tag-an-image-into-a-repository needs some updates

yongtang committed:
Update remote API docs for the removal of deprecated `force` in `docker tag`.

This fix updates remote API docs for the removal of deprecated `force` in `docker tag`.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @vdemeester. The related docs `docker_remote_api_v1.24.md` and `docker_remote_api.md` has been updated to include the changes in this pull request.

thaJeztah commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -122,6 +122,7 @@ This section lists each version from latest to oldest.  Each listing includes a
 * `GET /events` now supports a `reload` event that is emitted when the daemon configuration is reloaded.
 * `GET /events` now supports filtering by daemon name or ID.
 * `GET /images/json` now supports filters `since` and `before`.
+* `POST /images/(name)/tag` now removes deprecated query parameter `force`.
```
Thinking of better wording for this, perhaps;

```Markdown
* `POST /images/(name)/tag` no longer has a `force` query parameter. 
```

thaJeztah commented:
One small suggestion, but LGTM otherwise

yongtang commented:
Thanks @thaJeztah the pull request has been updated.

vdemeester commented:
LGTM :angel:

chriskuehl commented:
It seems that there's no backwards-compatible way to force-tag images now that works with Docker versions prior to 1.10 (released only 5 months ago). I've opened #24494 for discussion; please consider waiting to drop this flag, as this will make it very difficult to migrate large Docker deployments to 1.12.
